ui/packages/consul-ui/app/components/consul/intention/form/index.hbs

```diff
@@ -56,7 +56,10 @@ as |api|>
     </BlockSlot>
 
     <BlockSlot @name="form">
-{{#let api.data as |item|}}
+{{#let
+  api.data
+  (not (can 'write intention' item=api.data))
```

Comment:
This line here was the bugfix, previously the `readonly` variable just below on line 63 was undefined always 😬 

Most of the other change here is to test this so it doesn't happen again

Reply:
Maybe an opportunity to use [`cannot`](https://github.com/minutebase/ember-can#cannot)?

```suggestion
  (cannot 'write intention' item=api.data)
```

Reply:
In the codebase we prefer using `not (can "thing")`, that way folks always use the same method to flip the conditional no matter if it's using `can` or some other flippable helper. So I'll probably leave this one.

Reply:
Fair enough